src/components/SideNav.js

```diff
@@ -1,45 +1,73 @@
 import React from 'react'
 import i18n from '@dhis2/d2-i18n'
 import { Menu, MenuItem, MenuSectionHeader, MenuDivider } from '@dhis2/ui'
+import { useDataQuery } from '@dhis2/app-service-data'
 
 import MenuItemLink from './MenuItemLink.js'
 import { PATHS } from '../config'
 
-const SideNav = () => (
-    <nav>
-        <Menu>
-            <MenuItemLink to={PATHS.DASHBOARD} label={i18n.t('Dashboard')} />
-            <MenuSectionHeader label={i18n.t('Analysis')} />
-            <MenuItemLink
-                to={PATHS.OUTLIERS_AND_MISSING_DATA_ANALYSIS}
-                label={i18n.t('Outliers and missing data')}
-            />
-            <MenuItemLink
-                to={PATHS.CONSISTENCY_ANALYSIS}
-                label={i18n.t('Consistency')}
-            />
-            <MenuDivider />
-            <MenuItemLink
-                to={PATHS.ANNUAL_REPORT}
-                label={i18n.t('Annual report')}
-            />
-            <MenuSectionHeader label={i18n.t('More')} />
-            <MenuItemLink
-                to={PATHS.ADMINISTRATION}
-                label={i18n.t('Administration')}
-            />
-            <MenuItemLink
-                to={PATHS.DATA_EXPORT_FOR_EXCEL}
-                label={i18n.t('Data export for Excel tool')}
-            />
-            <MenuItem
-                href="https://www.ssb.no/en/helse/artikler-og-publikasjoner/manual-for-the-dhis2-quality-tool"
-                target="_blank"
-                label={i18n.t('User manual')}
-            />
-            <MenuDivider />
-        </Menu>
-    </nav>
-)
+const query = {
+    me: {
+        resource: 'me',
+    },
+}
+const adminAuthorities = ['ALL', 'F_INDICATOR_PUBLIC_ADD']
 
+const SideNav = () => {
+    const { loading, error, data } = useDataQuery(query)
+
+    if (loading) {
+        return null
+    }
+
+    let isAdmin = false
+    if (data && data.me && data.me.authorities) {
+        isAdmin =
+            data.me.authorities.filter(auth => adminAuthorities.includes(auth))
+                .length > 0
+    }
```

Comment:
We should be able to use optional chaining, and I think we should also use different array operators:
```javascript
const isAdmin = data?.me.authorities.some(userAuth =>
    adminAuthorities.includes(userAuth)
)
```
(fully untested)

In any case `.some` is better than `.filter` here because a) it returns a boolean, and b) it stops iterating once it finds a match.

Reply:
Tested it at jsbench.me and found that some is between 2-10x faster in my fictional cases, so I'll replace filter with some